### ISSUE TITLE
g_error_leak rule should also check for g_propagate_prefixed_error

### ISSUE THREAD

**Issue by phako** [bilelmoussaoui/gobject-lint#104]
Otherwise this is a false positive:

```c
{
        GError *inner_error = NULL;

        /* The 4th argument 'iface_name' can't be NULL even though Glib API doc says you
         * can. 'NULL' will fail the test.
         */
        if (!g_socket_join_multicast_group (priv->socket,
                                            group,
                                            FALSE,
                                            priv->device_name,  /*   e.g. 'lo' */
                                            &inner_error)) {
                gchar *address = g_inet_address_to_string (group);
                g_propagate_prefixed_error (error,
                                            inner_error,
                                            "Failed to join group %s",
                                            address);
                g_free (address);

                return FALSE;
        }

        return TRUE;
```
